workshop/1-Personalization/1.2-Real-Time-Events-Segment.ipynb

```diff
@@ -0,0 +1,451 @@
+    "        };\n",
+    "\n",
+    "        if (this.segmentEnabled()) {\n",
+    "            window.analytics.track('ProductViewed', eventProperties);\n",
+    "        }\n",
+    "```"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "source": [
+    "This allows you to collect data that is relevant to any tool that might need to be kept updated with the latest user behavior.  \n",
+    "\n",
+    "## Sending Real-Time Events\n",
+    "\n",
+    "Since you have already connected Segment to Personalize, let’s test this data path by triggering events from the Retail Demo Store user interface which is deployed in your AWS account.\n",
+    "\n",
+    "Run the following code to set the SSM parameter that holds the Segment write key."
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set the Segment write key in the string below, and run this cell.  \n",
+    "\n",
+    "# THIS IS ONLY REQUIRED IF YOU DID NOT SET THE SEGMENT WRITE KEY IN YOUR ORIGINAL DEPLOYMENT\n",
+    "\n",
+    "import boto3\n",
+    "\n",
+    "ssm = boto3.client('ssm')\n",
+    "\n",
+    "if segment_write_key:\n",
+    "    response = ssm.put_parameter(\n",
+    "        Name='retaildemostore-segment-write-key',\n",
+    "        Value='{}'.format(segment_write_key),\n",
+    "        Type='String',\n",
+    "        Overwrite=True\n",
+    "    )\n",
+    "\n",
+    "print(segment_write_key)"
+   ]
+  },
+  {
+   "source": [
+    "You now have an environment variable that will enable the Segment data collection library in the code in `index.html`.  All we need to do now, is force a re-deploy of the Retail Demo Store.  \n",
+    "\n",
+    "To do that, go back to your AWS Console tab or window, and select Code Pipeline from the Services search.  Then, find the pipeline name that contains `WebUIPipeline` and click that link.\n",
+    "\n",
+    "![Segment Destination Setup](images/segment/segment-step-25.png)\n",
+    "\n",
+    "Then, select the ‘Release Change’ button, and confirm the release once the popup shows up.  You will see a confirmation that the pipeline is re-deploying the web ui for the Retail Demo Store.\n",
+    "\n",
+    "![Segment Destination Setup](images/segment/segment-step-26.png)\n",
+    "\n",
+    "This process should complete in a few minutes.  Once this is done, you will see the bottom tile confirm that your deployment has completed:\n",
+    "\n",
+    "![Segment Destination Setup](images/segment/segment-step-27.png)\n",
+    "\n",
+    "## Log in as a Retail Demo Store User\n",
+    "\n",
+    "Once this is confirmed, tab back to the Retail Demo Store web UI, and refresh the screen to reload the user interface.  In the Personalize workshop, you created an account for the Retail Demo Store.  If you are not logged in as that account, log in now.\n",
```

Comment:
I suggest emphasizing this a little more since it's easy to miss and important to do before the next steps will work.